src/Eshopworld.WorkerProcess/Configuration/ServiceConfigurationExtensions.cs

```diff
@@ -16,55 +17,70 @@ namespace EShopworld.WorkerProcess.Configuration
     [ExcludeFromCodeCoverage]
     public static class ServiceConfigurationExtensions
     {
-        private const string DocumentClientServiceName = "WorkerProcessDocumentClient";
+        private const string WorkerLeaseOptions = "WorkerProcess:WorkerLease";
+        private const string CosmosDataStoreOptions = "WorkerProcess:CosmosDataStore";
 
         /// <summary>
-        ///     Adds a <see cref="IWorkerLease"/> to the services collection
+        ///  Adds a <see cref="IWorkerLease"/> to the services collection
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/></param>
-        public static void AddWorkerLease(this IServiceCollection services)
+        /// <param name="configuration">The <see cref="IConfigurationRoot"/></param>
+        public static void AddWorkerLease(this IServiceCollection services, IConfigurationRoot configuration)
         {
-            var serviceProvider = services.BuildServiceProvider();
-
-            var options = serviceProvider
-                .GetService<IOptionsMonitor<WorkerLeaseOptions>>();
+            services.AddOptions();
+            services.Configure<WorkerLeaseOptions>(configuration.GetSection(WorkerLeaseOptions));
 
-            if (options.CurrentValue == null)
-                throw new WorkerLeaseException($"[{typeof(IOptionsMonitor<WorkerLeaseOptions>)}] was not configured");
+            var cosmosDataStoreOptions = configuration.BindSection<CosmosDataStoreOptions>(CosmosDataStoreOptions,
+                m => { m.AddMapping(x => x.ConnectionString, "cm--cosmos-connection--worker-process"); });
```

Comment:
can we put the secret name in a constant and reuse it

Reply:
absolutely

Reply:
fixed